src/pin.js

```diff
@@ -0,0 +1,133 @@
+/* eslint-env mocha */
+/* eslint max-nested-callbacks: ["error", 8] */
+
+'use strict'
+
+const expect = require('chai').expect
+const fs = require('fs')
+const path = require('path')
+
+const testfile = fs.readFileSync(path.join(__dirname, './data/testfile.txt'))
+
+module.exports = (common) => {
+  describe('.pin', () => {
+    let ipfs
+
+    before(function (done) {
+      // CI is slow
+      this.timeout(20 * 1000)
+
+      common.setup((err, factory) => {
```

Comment:
It is the 'interface test suite pattern', see: https://github.com/ipfs/interface-ipfs-core#usage